metadata-ingestion/src/datahub/ingestion/source/state/redundant_run_skip_handler.py

```diff
@@ -127,22 +127,24 @@ def should_skip_this_run(self, cur_start_time_millis: int) -> bool:
         if not self.is_checkpointing_enabled() or self._ignore_old_state():
             return False
         # Determine from the last check point state
-        last_successful_pipeline_run_end_time_millis: Optional[int] = None
+        last_successful_pipeline_run_start_time_millis: Optional[int] = None
         last_checkpoint = self.state_provider.get_last_checkpoint(
             self.job_id, BaseUsageCheckpointState
         )
         if last_checkpoint and last_checkpoint.state:
             state = cast(BaseUsageCheckpointState, last_checkpoint.state)
-            last_successful_pipeline_run_end_time_millis = state.end_timestamp_millis
+            last_successful_pipeline_run_start_time_millis = (
+                state.begin_timestamp_millis
+            )
 
         if (
-            last_successful_pipeline_run_end_time_millis is not None
-            and cur_start_time_millis <= last_successful_pipeline_run_end_time_millis
+            last_successful_pipeline_run_start_time_millis is not None
+            and cur_start_time_millis <= last_successful_pipeline_run_start_time_millis
         ):
             warn_msg = (
-                f"Skippig this run, since the last run's bucket duration end: "
-                f"{get_datetime_from_ts_millis_in_utc(last_successful_pipeline_run_end_time_millis)}"
-                f" is later than the current start_time: {get_datetime_from_ts_millis_in_utc(cur_start_time_millis)}"
+                f"Skippig this run, since the last run's bucket duration start: "
+                f"{get_datetime_from_ts_millis_in_utc(last_successful_pipeline_run_start_time_millis)}"
+                f" is later than or equal to the current start_time: {get_datetime_from_ts_millis_in_utc(cur_start_time_millis)}"
             )
```

Comment:
1. Redundant run skip handler is only useful for scheduled executions with default start and end times
2. For scheduled executions  - the comparison of start_time logic works great as it would be exactly same for all runs on same UTC day (with default start, end times), leading to skipping redundant usage runs. When current UTC day changes, start_time changes, hence the new start_time will be later than the last run's start_time.